src/main/java/bio/terra/service/snapshotbuilder/utils/HierarchyQueryBuilder.java

```diff
@@ -92,73 +99,69 @@ public Query generateQuery(SnapshotBuilderDomainOption domainOption, int concept
    * href="https://www.ohdsi.org/web/wiki/doku.php?id=documentation:vocabulary:standard_classification_and_source_concepts">Standard,
    * Classification, and Source Concepts</a>
    */
-  private static BinaryFilterVariable requireStandardConcept(TableVariable concept) {
+  private static BinaryFilterVariable requireStandardConcept(SourceVariable concept) {
     return BinaryFilterVariable.equals(
         concept.makeFieldVariable(Concept.STANDARD_CONCEPT), new Literal("S"));
   }
 
-  /**
-   * Given a concept ID, select all of its parent concept IDs. Note that a concept may be its own
-   * ancestor so that case must be excluded.
-   *
-   * <pre>
-   * {@code SELECT ancestor_concept_id FROM concept_ancestor
-   * WHERE descendant_concept_id = :conceptId: AND ancestor_concept_id != :conceptId:;}
-   * </pre>
-   */
-  private Query selectAllParents(int conceptId) {
-    var conceptAncestor =
-        TableVariable.forPrimary(TablePointer.fromTableName(ConceptAncestor.TABLE_NAME));
-    var conceptIdLiteral = new Literal(conceptId);
-    FieldVariable ancestorConceptId =
-        conceptAncestor.makeFieldVariable(ConceptAncestor.ANCESTOR_CONCEPT_ID);
-    return new Query(
-        List.of(ancestorConceptId),
-        List.of(conceptAncestor),
-        BooleanAndOrFilterVariable.and(
-            BinaryFilterVariable.equals(
-                conceptAncestor.makeFieldVariable(ConceptAncestor.DESCENDANT_CONCEPT_ID),
-                conceptIdLiteral),
-            BinaryFilterVariable.notEquals(ancestorConceptId, conceptIdLiteral)));
+  static SourceVariable joinToFilterConcepts(FieldVariable parentId, int conceptId) {
```

Comment:
Maybe rename to something like joinParents or joinChildren - or what does this method do exactly? Maybe a test for this would be beneficial for terms of documentation?

Reply:
Great point - I've added comments and tests for all of these methods.